### PR TITLE
Sawn-off Shotgun Invisibility Fix

### LINKED ISSUE #12032: Sawn off double barrel shotgun has no sprite
if you use a circular saw on bartender's double barrel shotgun for some strange reason it becomes invisible.

### PR DESCRIPTION
**Pull request by IHadMyUsernameTaken**
## Changelog
:cl:
fix: All sawn-off shotguns are visible again.
/:cl:

## About The Pull Request
The change done in #12024 made all shotguns except the riot shotgun (doesn't apply to contenders either, but we don't consider them here) use the projectile.dmi file with the newer sprites.
Due to this change, only the sawn-off riot shotgun would have a visible sprite; the file with the new sprites doesn't have any icon state for sawn-off shotguns such that they would have "-sawn" in their name, which was assumed to be true for all sawn-off guns in the update_icon proc of ballistic guns.

I made all shotguns use the projectile.dmi file with the old sprites for their icon states, since to me it seemed that the change I mentioned above wasn't intentional, and make the update_icon proc use "_sawn" to form the name of the sawn-off weapon's icon state for when someone either intentionally or not happens to make a shotgun use the other projectile.dmi file; icon states of the sawn-off shotguns in the Hippie station projectiles.dmi file had to be renamed to use "_sawn" to take this change into account.

Fixes #12032.

## Why It's Good For The Game
This should make all sawn-off shotguns have visible sprites regardless of which projectiles.dmi file you decide to use and make them use their old sprites again, since the change to the newer ones most likely wasn't intended.

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -72,9 +72,9 @@
 		return
 	..()
 	if(current_skin)
-		icon_state = "[unique_reskin[current_skin]][sawn_off ? "-sawn" : ""]" // hippie -- fixes sawn-off sprites
+		icon_state = "[unique_reskin[current_skin]][sawn_off ? "_sawn" : ""]"
```
Comment (yoyobatty) [inline code/modules/projectiles/guns/ballistic.dm]: Leave the //hippie edit in

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -72,9 +72,9 @@
...
 	else
-		icon_state = "[initial(icon_state)][suppressed ? "-suppressed" : ""][sawn_off ? "-sawn" : ""]"
+		icon_state = "[initial(icon_state)][suppressed ? "-suppressed" : ""][sawn_off ? "_sawn" : ""]"
```
Comment (yoyobatty) [inline code/modules/projectiles/guns/ballistic.dm]: Add //hippie edit

**Review by yoyobatty — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (IHadMyUsernameTaken) [reply code/modules/projectiles/guns/ballistic.dm]: The comments should be added now.

**Review by yoyobatty — APPROVED** [review]
You sure this doesn't break anything?

**Comment by IHadMyUsernameTaken** [comment]
What I meant to say was that it should work if my previous text seemed to say otherwise (which was what I thought at least, so here's an edit). I tried it again today and didn't see any problems that would specifically stem from these changes.

**Pull request merged**